worker/src/handlers/batchStatementDeletion/batchStatementDeletion.js

```diff
@@ -0,0 +1,125 @@
+import boolean from 'boolean';
+import { get, map } from 'lodash';
+import BatchDelete, { asTime } from 'lib/models/batchDelete';
+import parseQuery from 'lib/helpers/parseQuery';
+import Statement from 'lib/models/statement';
+import { publish as pubishToQueue } from 'lib/services/queue';
+import { BATCH_STATEMENT_DELETION_QUEUE } from 'lib/constants/batchDelete';
+import { SITE_SETTINGS_ID } from 'lib/constants/siteSettings';
+import SiteSettings from 'lib/models/siteSettings';
+import Client from 'lib/models/client';
+import getScopeFilter from 'lib/services/auth/filters/getScopeFilter';
+import moment from 'moment';
+
+export default async ({
+  batchDeleteId,
+  publish = pubishToQueue
+}, jobDone) => {
+  if (boolean(get(process.env, 'ENABLE_BATCH_STATEMENT_DELETION', false)) === false) {
+    // Clear the job and don't do anything
+    jobDone();
+    return;
+  }
+
+  const siteSettings = await SiteSettings.findOne({
+    _id: SITE_SETTINGS_ID
+  });
+  if (
+    get(siteSettings, 'batchDeleteWindowStartTime') &&
+    get(siteSettings, 'batchDeleteWindowDuration') &&
+    !asTime(moment()).isBetween(
+      asTime(get(siteSettings, 'batchDeleteWindowStartTime')),
+      asTime(get(siteSettings, 'batchDeleteWindowStartTime'))
+        .add(get(siteSettings, 'batchDeleteWindowDuration'), 'milliseconds')
+    )
+  ) {
+    // We're not in the window
+    jobDone();
+    return;
+  }
+
+  const batchDelete = await BatchDelete.findOneAndUpdate({
+    _id: batchDeleteId,
+    processing: false,
+    done: false
+  }, {
+    processing: true
+  });
+
+  if (!batchDelete) {
+    // Another worker is proccessing this job, or it has finished
+    jobDone();
+    return;
+  }
+
+  let scopeFilter;
+  try {
+    scopeFilter = await getScopeFilter({
+      modelName: 'statement',
+      actionName: 'delete',
+      authInfo: {
+        client: await Client.findOne({ _id: batchDelete.client })
+      }
+    });
+  } catch (err) {
+    if (err.message === 'Priviliges not sufficient for this operation') {
+      // Do nothing
+      await BatchDelete.findOneAndUpdate({
+        _id: batchDeleteId
+      }, {
+        processing: false,
+        done: true
+      });
+
+      jobDone();
+      return;
+    }
+    throw err;
+  }
+
+  const filter = {
+    ...(await parseQuery(batchDelete.filter)),
+    ...scopeFilter
+  };
+
+  const docs = await Statement.aggregate()
```

Comment:
Why did you use `aggregate` instead of `find` here @cbishopvelti?

Reply:
@ht2 Chris thinks the aggregate performs better than the find. He had also thought that the find didn't take a projection, but I've shown him that [it does take a find in the docs](https://mongoosejs.com/docs/api.html#model_Model.find).

Reply:
See here for a good example where we use a cursor on a filtered query with a lean projection of just _id: https://github.com/LearningLocker/enterprise/blob/master/lib/services/journey/getJourneyStatements.js#L18

Reply:
Ok thank you 👍